build/compiler.py

```diff
@@ -222,6 +222,59 @@ def generate(self, force=False):
     return True
 
 
+class TypescriptGenerator(object):
+  def __init__(self, source_files, build_name):
+    self.source_files = _canonicalize_source_files(source_files)
+    self.output = _get_source_path('dist/' + build_name + '.d.ts')
+
+  def generate(self, force=False):
+    """Generates TypeScript definitions for the files in |self.source_files|.
+
+    Args:
+      force: Generate the output even if the inputs have not changed.
+
+    Returns:
+      True on success; False on failure.
+    """
+    if not force and not _must_build(self.output, self.source_files):
+      return True
+
+    ts_generator = _get_source_path('build/typescript/main.ts')
+    ts_node = _get_source_path('node_modules/ts-node/dist/bin')
+    tsconfig = _get_source_path('build/typescript/tsconfig.json')
+
+    cmd_line = ['node', ts_node, '-P', tsconfig, ts_generator, '--output', self.output]
```

Comment:
We have utilities for this to make sure everything works on Linux, Mac, Windows, and Cygwin.  For example, see below for "lessc":

```py
lessc = shakaBuildHelpers.get_node_binary('less', 'lessc')
# ...
cmd_line = lessc + less_options + [self.main_source_file, self.output]
```

Please note that get_node_binary returns a list of strings, not a single string as you might guess.